rust/src/main.rs

```diff
@@ -11,19 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-use std::io;
 
 #[macro_use]
 extern crate log;
 extern crate chrono;
 extern crate fern;
 
+extern crate dirs;
+
 extern crate xi_core_lib;
 extern crate xi_rpc;
 
+use std::io;
+use std::fs;
+use std::path::PathBuf;
+
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
+const XI_DIRECTORY: &'static str = "xi-core";
+const XI_LOG_FILE: &'static str = "xi-core.log";
+
+fn get_logging_directory() -> Option<PathBuf> {
+    if let Some(mut log_dir) = dirs::data_local_dir() {
+        log_dir.push(XI_DIRECTORY);
+        Some(log_dir)
+    } else {
+        eprintln!("[WARNING] The dir library was not able to find a directory for this platform");
+        None
+    }
+}
+
+fn path_for_log_file<P: AsRef<str>>(filename: P) -> Option<PathBuf> {
+    if let Some(mut logging_directory) = get_logging_directory() {
```

Comment:
just an observation: you could write this `let mut logging_directory = get_logging_directory()?;`, and the `?` will just return `None` if the result is `None`, and you can continue without the `{ .. }` block.

Reply:
I've given this a go & swapped the Options back to Results as I wanted it to respond with an io Not Found error.
I've tried forcing `get_logging_directory` to always respond with the error; it was caught fine (by the match on line 79) & didn't cause any exit or crash but still gives some valuable info.